src/main/java/au/org/emii/ncdfgenerator/NcdfEncoderBuilder.java

```diff
@@ -0,0 +1,1476 @@
+
+
+package au.org.emii.ncdfgenerator;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream ;
+import java.io.OutputStream ;
+import java.io.IOException;
+
+import java.nio.charset.Charset;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern ;
+import java.util.regex.Matcher;
+import java.util.AbstractMap.SimpleImmutableEntry;
+
+import java.sql.*;
+
+import java.text.SimpleDateFormat;
+
+import ucar.nc2.NetcdfFileWriteable;
+import ucar.nc2.Dimension;
+import ucar.ma2.DataType;
+import ucar.ma2.Array;
+import ucar.ma2.ArrayDouble;
+import ucar.ma2.Index;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.NodeList;
+
+import java.nio.charset.StandardCharsets;
+import javax.xml.parsers.DocumentBuilderFactory;
```

Comment:
Perhaps use an IDE to identify issues with current code - e.g. unused imports, missing serialVersionUID field's etc.  Findbugs is also good